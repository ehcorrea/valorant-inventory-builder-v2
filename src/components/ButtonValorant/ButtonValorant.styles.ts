import styled, { css } from 'styled-components/native';

export const Container = styled.View`
  ${({ theme }) => css`
    border: solid 1px ${theme.colors.white};
    height: ${theme.spacings.vertical.huge}px;
    padding: ${theme.spacings.xxxsmall}px;
    width: 100%;
  `}
`;

export const Wrapper = styled.View`
  ${({ theme }) => css`
    align-items: center;
    background-color: ${theme.colors.red};
    flex: 1;
    justify-content: center;
  `}
`;
