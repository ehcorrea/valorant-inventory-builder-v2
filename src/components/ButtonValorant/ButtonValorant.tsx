import { TouchableOpacity, TouchableOpacityProps } from 'react-native';

import * as S from './ButtonValorant.styles';

export type ButtonValorantTypes = TouchableOpacityProps;

export function ButtonValorant({ children, ...props }: ButtonValorantTypes) {
  return (
    <TouchableOpacity {...props}>
      <S.Container>
        <S.Wrapper>{children}</S.Wrapper>
      </S.Container>
    </TouchableOpacity>
  );
}
