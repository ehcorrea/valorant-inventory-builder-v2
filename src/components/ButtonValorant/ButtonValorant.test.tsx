import { Text } from 'react-native';

import { fireEvent, render } from '@/test/utils';

import { ButtonValorant } from './ButtonValorant';

describe('<ButtonValorant/>', () => {
  const spy = jest.fn();

  test('should render ButtonValorant with success', async () => {
    const container = render(
      <ButtonValorant testID="ButtonValorant" onPress={spy}>
        <Text>Render Text</Text>
      </ButtonValorant>
    );

    const button = await container.findByTestId('ButtonValorant');
    fireEvent.press(button);

    expect(container.getByText('Render Text')).toBeTruthy();
    expect(spy).toHaveBeenCalled();

    expect(container).toMatchSnapshot();
  });
});
