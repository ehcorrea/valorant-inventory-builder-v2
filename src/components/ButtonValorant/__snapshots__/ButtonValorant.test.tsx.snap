// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonValorant/> should render ButtonValorant with success 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="ButtonValorant"
>
  <View
    style={
      {
        "borderColor": "#FFFBF5",
        "borderStyle": "solid",
        "borderWidth": 1,
        "height": 102,
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FD4556",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text>
        Render Text
      </Text>
    </View>
  </View>
</View>
`;
